dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -0,0 +1,1068 @@
+  DynamicType_rch elem_type = descriptor.element_type;
+  if (elem_type->get_kind() == TK_ALIAS) {
+    elem_type = get_base_type(elem_type);
+  }
+
+  ACE_CDR::ULong primitive_size = 0;
+  if (is_primitive(elem_type, primitive_size)) {
+    ACE_CDR::ULong length;
+    if (!(strm_ >> length)) {
+      if (DCPS_debug_level >= 10) {
+        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) DynamicData::skip_sequence_member -")
+                   ACE_TEXT(" Failed to deserialize a primitive sequence member\n")));
+      }
+      return false;
+    }
+
+    return skip("skip_sequence_member", "Failed to skip a primitive sequence member",
+                length, primitive_size)
+  } else {
+    return skip_collection_member(TK_SEQUENCE);
+  }
+}
+
+bool DynamicData::skip_array_member(DynamicType_rch array_type)
+{
+  TypeDescriptor descriptor;
+  array_type->get_descriptor(descriptor);
+  DynamicType_rch elem_type = descriptor.element_type;
+  if (elem_type->get_kind() == TK_ALIAS) {
+    elem_type = get_base_type(elem_type);
+  }
+
+  ACE_CDR::ULong primitive_size = 0;
+  if (is_primitive(elem_type, primitive_size)) {
+    const LBoundSeq& bounds = descriptor.bound;
+    ACE_CDR::ULong num_elems = 1;
+    for (unsigned i = 0; i < bounds.size(); ++i) {
+      num_elems *= bounds[i];
+    }
+
+    return skip("skip_array_member", "Failed to skip a primitive array member",
+                num_elems, primitive_size)
+  } else {
+    return skip_collection_member(TK_ARRAY);
+  }
+}
+
+bool DynamicData::skip_map_member(DynamicType_rch map_type)
+{
+  TypeDescriptor descriptor;
+  map_type->get_descriptor(descriptor);
+
+  DynamcType_rch elem_type = descriptor.element_type;
+  if (elem_type->get_kind() == TK_ALIAS) {
+    elem_type = get_base_type(elem_type);
+  }
+
+  DynamicType_rch key_type = descriptor.key_element_type;
+  if (key_type->get_kind() == TK_ALIAS) {
+    key_type = get_base_type(key_type);
+  }
+
+  ACE_CDR::ULong key_primitive_size = 0, elem_primitive_size = 0;
+  if (is_primitive(key_type, key_primitive_size) &&
+      is_primitive(elem_type, elem_primitive_size)) {
+    ACE_CDR::ULong length;
+    if (!(strm_ >> length)) {
+      if (DCPS_debug_level >= 10) {
+        ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) DynamicData::skip_map_member -")
+                   ACE_TEXT(" Failed to deserialize length of a primitive map member\n")));
+      }
+      return false;
+    }
+
+    for (unsigned i = 0; i < length; ++i) {
+      if (!skip("skip_map_member", "Failed to skip a key of a primitive map member",
+                1, key_primitive_size)) {
+        return false;
+      }
+      if (!skip("skip_map_member", "Failed to skip an element of a primitive map member",
+                1, elem_primitive_size)) {
+        return false;
+      }
+    }
+    return true;
+  } else {
+    return skip_collection_member(TK_MAP);
+  }
+}
+
+bool DynamicData::skip_collection_member(TypeKind kind)
+{
+  const char* kind_str = 0;
+  if (kind == TK_SEQUENCE) {
```

Comment:
case? Maybe use a helper function for the string translation just like we have for retcode?